src/mpi/comm/comm_split_type.c

```diff
@@ -497,6 +619,65 @@ int MPIR_Comm_split_type_node_topo(MPIR_Comm * user_comm_ptr, int split_type, in
     goto fn_exit;
 }
 
+#undef FUNCNAME
+#define FUNCNAME MPIR_Comm_split_type_node_topo
+#undef FCNAME
+#define FCNAME MPL_QUOTE(FUNCNAME)
+int MPIR_Comm_split_type_network_topo(MPIR_Comm * comm_ptr, char *hintval, int key,
+                                      MPIR_Comm ** newcomm_ptr)
+{
+    int hintval_size, hintval_size_global, mpi_errno, info_args_are_equal;
+    char *hintval_global;
+    if (hintval == NULL) {
+        strcpy(hintval, "");
+    }
+
+    /* Compare hintval string length */
+    hintval_size = strlen(hintval);
+    mpi_errno = compare_info_args(&hintval_size, &hintval_size_global,
+                                  sizeof(int) / sizeof(char), comm_ptr, &info_args_are_equal);
+    if (mpi_errno)
+        MPIR_ERR_POP(mpi_errno);
+
+    /* if all processes do not give the same length of hintval, skip
+     * topology-aware comm split */
+    if (!info_args_are_equal)
+        goto use_node_comm;
+
+    hintval_global = (char *) MPL_malloc(sizeof(hintval), MPL_MEM_OTHER);
+    /* Compare the hintval strings */
+    mpi_errno = compare_info_args(hintval, hintval_global,
+                                  (strlen(hintval) + 1) * sizeof(char), comm_ptr,
+                                  &info_args_are_equal);
+
+    if (hintval_global != NULL)
+        MPL_free(hintval_global);
```

Comment:
The checks to make sure the info hints are the same for all processes are being duplicated.  They need to be consolidated into one place.